config_parser.js

```diff
@@ -1,21 +1,41 @@
 const fs = require('fs');
 const Ajv = require('ajv');
+const path = require('path');
 
 const ajv = new Ajv({ allErrors: true, useDefaults: true });
 
-function loadConfig(configPath, schemaPath) {
-    const schema = JSON.parse(fs.readFileSync(schemaPath, 'utf-8'));
-    const config = JSON.parse(fs.readFileSync(configPath, 'utf-8'));
+function loadConfig(configPath, schemaPath, defaultValues = {}, preprocess = null) {
+    try {
+        console.log(`Loading schema from ${schemaPath}...`);
+        const schema = JSON.parse(fs.readFileSync(schemaPath, 'utf-8'));
+        console.log('Schema loaded successfully.');
 
-    const validate = ajv.compile(schema);
-    const valid = validate(config);
+        console.log(`Loading config from ${configPath}...`);
+        const config = JSON.parse(fs.readFileSync(configPath, 'utf-8'));
+        console.log('Config loaded successfully.');
 
-    if (!valid) {
-        console.log('Validation errors:', validate.errors);
-        throw new Error('Config validation failed');
-    }
+        const finalConfig = { ...defaultValues, ...config };
+
+        const validate = ajv.compile(schema);
```

Comment:
It's a great idea to make a feature with validation, think of another way to update the function.
![photo_2024-06-12_20-24-29](https://github.com/VadimOpt/Lab4_MTRPZ/assets/105482228/0b5f3908-a71e-4218-87a2-96c67c22bd37)